### DISCUSSION TITLE
DNS check failed

### DISCUSSION THREAD

**Discussion by MyUncleSam** [anonaddy/anonaddy#535] [Q&A] [answered]
Hi there,

I added a custom domain as user in my self hosted anonaddy. I did all DNS configuration as told but it is still complaining.

Right now it complains about my "CNAME ._domainkey record not found".

I tried to configure as told using the CNAME. Also I tried to add it using the TXT record as the main domain. But without any changes.

DNS record:
`default._domainkey		IN	CNAME	default._domainkey.mydomain.tld.`

**Comment by willbrowningme** [comment, marked as answer]
You need to make sure you have set "default" as the value for ANONADDY_DKIM_SELECTOR in your .env file.

**Reply by MyUncleSam** [reply]
thx that worked :-)

**Comment by zkazsi** [comment]
This was also my problem, and the setting has solved it (domain check now passes, BUT:

now (after setting this up this morning) **I receive NO mails at all** for my secondary domain :-(

For all my mails I get a notification from addy about "New failed delivery on AnonAddy". 
The failed deliveries page refers me to the logs, where I find this:

`anonaddy        | Oct 29 06:58:04 app postfix/pipe[1703]: 5E09F4038F: to=<aliasexample@example.com>, relay=anonaddy, delay=2.6, delays=0.95/0.02/0/1.6, dsn=5.3.0, status=bounced (Command died with status 1: "php". Command output:  In DkimSigner.php line 42:                                                                                   Symfony\Component\Mime\Crypto\DkimSigner::__construct(): Argument #1 ($pk)      must be of type string, null given, called in /var/www/anonaddy/app/CustomM     ailDriver/CustomMailer.php on line 91`
 
I'm using version 0.14.1, with docker compose
It seems, I can still receive mails sent to the primary alias domain (where addy was set up initially, app.primaryalias.com)

Any ideas? I would appreciate any help.

**Reply by zkazsi** [reply]
Well, it took quite a bit of trial & error, but the solution is: 

(In addition to ANONADDY_DKIM_SELECTOR) I had to add also ANONADDY_DKIM_SIGNING_KEY to the docker-compose file: this is the 'path to the private DKIM signing key to be used to sign emails for custom domains'

It's actually all documented, you (I) just have to read it all.
I hope it will keep working
